pandas/tests/indexing/test_indexing.py

```diff
@@ -4250,7 +4250,8 @@ def test_series_partial_set_period(self):
                 pd.Period('2011-01-03', freq='D')]
         exp = Series([np.nan, 0.2, np.nan],
                      index=pd.PeriodIndex(keys, name='idx'), name='s')
-        assert_series_equal(ser.loc[keys], exp, check_index_type=True)
+        result = ser.loc[keys]
+        assert_series_equal(result, exp, check_index_type=True)
 
```

Comment:
you don't need the ``check_index_dtype``, this defaults to ``equiv`` which just equates ``Int64/RangeIndex``, e.g. exact equivalents.